comit/src/network/protocols/orderbook.rs

```diff
@@ -357,4 +452,148 @@ impl Orderbook {
             bincode::serialize(&Message::TradingPair(trading_pair)).unwrap(),
         );
     }
+
+    fn poll(
+        &mut self,
+        _: &mut Context<'_>,
+        _: &mut impl PollParameters,
+    ) -> Poll<
+        NetworkBehaviourAction<
+            EitherOutput<GossipsubRpc, RequestProtocol<TakeOrderCodec>>,
+            BehaviourOutEvent,
+        >,
+    > {
+        if let Some(event) = self.events.pop_front() {
+            return Poll::Ready(NetworkBehaviourAction::GenerateEvent(event));
+        }
+
+        Poll::Pending
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        asset::{self, Erc20, Erc20Quantity},
+        network::test::{await_events_or_timeout, connect, new_swarm},
+    };
+    use atty::{self, Stream};
+    use libp2p::Swarm;
+    use log::LevelFilter;
+    use spectral::prelude::*;
+    use std::str::FromStr;
+    use tracing::{info, subscriber, Level};
+    use tracing_log::LogTracer;
+    use tracing_subscriber::FmtSubscriber;
+
+    fn create_order(id: PeerId, maker_addr: Multiaddr) -> Order {
+        Order::new(id, NewOrder {
+            buy: asset::Bitcoin::from_sat(100),
+            sell: Erc20 {
+                token_contract: Default::default(),
+                quantity: Erc20Quantity::max_value(),
+            },
+            absolute_expiry: 100,
+            maker_addr,
+        })
+    }
+
+    fn refund_redeem_identities() -> (crate::bitcoin::Address, identity::Ethereum) {
+        let refund = crate::bitcoin::Address::from_str("2MufZ6LLCqYTvZnCwzfAjSKn26Xcnr19PBE")
+            .expect("failed to parse bitcoin address");
+
+        let redeem = identity::Ethereum::random();
+
+        (refund, redeem)
+    }
+
+    #[tokio::test]
+    async fn take_order_request_confirmation() {
+        let comit_level = LevelFilter::Debug;
+        let upstream_level = LevelFilter::Info;
+        init_tracing(comit_level, upstream_level).expect("failed to init tracing");
+
+        // arrange
+
+        let delay = Duration::from_secs(2);
+
+        let (mut alice_swarm, _, _alice_peer_id) = new_swarm(Orderbook::new);
+        let (mut bob_swarm, bob_addr, bob_peer_id) = new_swarm(Orderbook::new);
+        connect(&mut alice_swarm, &mut bob_swarm).await;
```

Comment:
Seeing this repeated, I got the idea of a "make_alice_and_bob" function that connects the two swarms and hands them back to you :)

Reply:
Yes, good idea. Will do when fixing the log initialization stuff.

Reply:
Make it separate from the log init stuff please.

Reply:
Added the new function and just deleted all the trace initialization stuff.